src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -140,48 +157,14 @@ public async Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
                 return !status.IsInProgress;
             }
 
-            private async Task<IVsOperationProgressStageStatusForSolutionLoad> GetProgressStageStatusAsync(CancellationToken cancellationToken)
-            {
-                var service = await _serviceProvider.GetServiceAsync<SVsOperationProgress, IVsOperationProgressStatusService>(throwOnFailure: false)
-                                                    .WithCancellation(cancellationToken).ConfigureAwait(false);
-
-                return service?.GetStageStatusForSolutionLoad(CommonOperationProgressStageIds.Intellisense);
-            }
-
-            private async Task EnsureInitializationAsync(CancellationToken cancellationToken)
+            private async ValueTask<IVsOperationProgressStageStatusForSolutionLoad?> GetProgressStageStatusAsync(CancellationToken cancellationToken)
             {
-                using (await _initializationGate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
+                if (_progressStageStatus.Task.IsCompleted)
```

Comment:
📝 This branch can be removed if https://github.com/microsoft/vs-threading/pull/696 is approved.